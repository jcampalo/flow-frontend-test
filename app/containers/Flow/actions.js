import { actionTypes as at } from './constants';

export const addRule = () => ({
  type: at.ADD_RULE
});

export const changeRule = ({ name, value, index }) => ({
  type: at.CHANGE_RULE,
  payload: { name, value, index }
});

export const changeContent = ({ value }) => ({
  type: at.CHANGE_CONTENT,
  payload: { value }
});

export const execute = () => ({
  type: at.EXECUTE
});

export const executeEnd = ({ results }) => ({
  type: at.EXECUTE_END,
  payload: { results }
});
