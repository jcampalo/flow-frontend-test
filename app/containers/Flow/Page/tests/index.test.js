import React from 'react';
import { create } from 'react-test-renderer';
import { IntlProvider } from 'react-intl';

import { FlowPage, mapDispatchToProps } from '../index';
import { addRule, changeRule } from '../../actions';

jest.mock('containers/Flow/ExecuteForm', () => 'ExecuteForm');

const initial = {
  results: [],
  rules: [],
  onInit: jest.fn()
};
const renderComponent = props => {
  const finalProps = { ...initial, ...props };

  return (
    <IntlProvider {...props}>
      <FlowPage {...finalProps} />
    </IntlProvider>
  );
};

describe('<FlowPage />', () => {
  it('should render component with no results and rules', () => {
    const tree = create(renderComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with rules', () => {
    const props = {
      rules: [{
        id: 'id',
        index: 0,
        title: 'title',
        body: 'body',
        rulePass: 'rulePass',
        ruleFail: 'ruleFail'
      }]
    };
    const tree = create(renderComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with results', () => {
    const props = {
      results: [{
        id: 0,
        result: true,
        title: 'title'
      }, {
        id: 2,
        result: false,
        title: 'title'
      }, {
        id: 3,
        title: 'title'
      }]
    };
    const tree = create(renderComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    describe('onAdd', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onAdd).toBeDefined();
      });

      it('should dispatch addRule when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);

        result.onAdd();

        expect(dispatch).toHaveBeenCalledWith(addRule());
      });
    });

    describe('onChangeRule', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);

        expect(result.onChangeRule).toBeDefined();
      });

      it('should dispatch changeRule when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        const value = 'value';
        const name = 'name';
        const index = 'index';

        result.onChangeRule({ name, value, index });

        expect(dispatch).toHaveBeenCalledWith(changeRule({ name, value, index }));
      });
    });
  });
});
