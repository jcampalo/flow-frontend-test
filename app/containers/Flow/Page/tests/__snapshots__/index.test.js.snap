// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlowPage /> should render component with no results and rules 1`] = `
<div
  className="Wrapper__WrapperExecute-jRgSbj ktmQR"
>
  <div
    className="WrapperTasks__WrapperTasks-cqWWRV htqoqU"
  >
    <div
      className="Wrapper__Wrapper-cZxMuO eDyakP"
    >
      <button
        className="StyledButton__StyledButton-iQSVZX fjLvae"
        onClick={undefined}
        type="button"
      >
        Add
      </button>
    </div>
  </div>
  <div
    className="WrapperExecute__WrapperExecute-gyaoLD dkjpTk"
  >
    <ExecuteForm />
    <span>
      Results
    </span>
  </div>
</div>
`;

exports[`<FlowPage /> should render component with results 1`] = `
<div
  className="Wrapper__WrapperExecute-jRgSbj ktmQR"
>
  <div
    className="WrapperTasks__WrapperTasks-cqWWRV htqoqU"
  >
    <div
      className="Wrapper__Wrapper-cZxMuO eDyakP"
    >
      <button
        className="StyledButton__StyledButton-iQSVZX fjLvae"
        onClick={undefined}
        type="button"
      >
        Add
      </button>
    </div>
  </div>
  <div
    className="WrapperExecute__WrapperExecute-gyaoLD dkjpTk"
  >
    <ExecuteForm />
    <span>
      Results
    </span>
    <div
      className="Wrapper-dyuznp iDoJpU"
    >
      <label
        className="Label-ghyrbC gTecle"
      >
        Rule title pass
      </label>
    </div>
    <div
      className="Wrapper-dyuznp iDoJpU"
    >
      <label
        className="Label-ghyrbC dLBDCo"
      >
        Rule title fail
      </label>
    </div>
    <div
      className="Wrapper-dyuznp iDoJpU"
    >
      <label
        className="Label-ghyrbC gTecle"
      >
        title
      </label>
    </div>
  </div>
</div>
`;

exports[`<FlowPage /> should render component with rules 1`] = `
<div
  className="Wrapper__WrapperExecute-jRgSbj ktmQR"
>
  <div
    className="WrapperTasks__WrapperTasks-cqWWRV htqoqU"
  >
    <div
      className="Wrapper-jVvkfh fasrFL"
    >
      <form
        className="Form-gTeswK dyrSex"
      >
        <div
          className="InputWrapper-bpDXXy bZGuau"
        >
          <label
            htmlFor="title"
          >
            <span
              className="Wrapper-hPVDJR hvMvQN"
            >
              <label
                className="Label-jdylQG fQrKdz"
              >
                Title
              </label>
              <input
                autoComplete="off"
                className="StyledInput__Input-eYzgai zNgEU"
                name="title"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="text"
                value="title"
              />
            </span>
          </label>
        </div>
        <div
          className="InputWrapper-bpDXXy bZGuau"
        >
          <label
            htmlFor="id"
          >
            <span
              className="Wrapper-hPVDJR hvMvQN"
            >
              <label
                className="Label-jdylQG fQrKdz"
              >
                Id
              </label>
              <input
                autoComplete="off"
                className="StyledInput__Input-eYzgai zNgEU"
                name="id"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="number"
                value="id"
              />
            </span>
          </label>
        </div>
        <div
          className="InputWrapper-bpDXXy bZGuau"
        >
          <label
            htmlFor="body"
          >
            <span
              className="Wrapper-ddcghJ kYIiWq"
            >
              <label
                className="Label-fIbAhr ffJDjh"
              >
                Body
              </label>
              <textarea
                autoComplete="off"
                className="StyledTextArea__TextArea-cqmHOB jGwssY"
                name="body"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                rows={6}
                type="text"
                value="body"
              />
            </span>
          </label>
        </div>
        <div
          className="InputWrapper-bpDXXy bZGuau"
        >
          <label
            htmlFor="rulePass"
          >
            <span
              className="Wrapper-hPVDJR hvMvQN"
            >
              <label
                className="Label-jdylQG fQrKdz"
              >
                If rule pased
              </label>
              <input
                autoComplete="off"
                className="StyledInput__Input-eYzgai zNgEU"
                name="rulePass"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="number"
                value="rulePass"
              />
            </span>
          </label>
        </div>
        <div
          className="InputWrapper-bpDXXy bZGuau"
        >
          <label
            htmlFor="ruleFail"
          >
            <span
              className="Wrapper-hPVDJR hvMvQN"
            >
              <label
                className="Label-jdylQG fQrKdz"
              >
                If rule failed
              </label>
              <input
                autoComplete="off"
                className="StyledInput__Input-eYzgai zNgEU"
                name="ruleFail"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="number"
                value="ruleFail"
              />
            </span>
          </label>
        </div>
      </form>
    </div>
    <div
      className="Wrapper__Wrapper-cZxMuO eDyakP"
    >
      <button
        className="StyledButton__StyledButton-iQSVZX fjLvae"
        onClick={undefined}
        type="button"
      >
        Add
      </button>
    </div>
  </div>
  <div
    className="WrapperExecute__WrapperExecute-gyaoLD dkjpTk"
  >
    <ExecuteForm />
    <span>
      Results
    </span>
  </div>
</div>
`;
